src/metabase/lib/schema.cljc

```diff
@@ -42,7 +42,12 @@
 (mr/def ::stage.native
   [:and
    [:map
-    {:decode/normalize common/normalize-map}
+    {:decode/normalize #(->> %
+                             common/normalize-map
+                             ;; filter out null :collection keys -- see #59675
+                             (m/filter-kv (fn [k v]
+                                            (not (and (= k :collection)
+                                                      (nil? v))))))}
```

Comment:
Optional, but this could be done with `(u/assoc-dissoc m :collection (:collection m))`.